linkerd/tls/src/main/scala/io/buoyant/linkerd/tls/TlsUtils.scala

```diff
@@ -22,11 +24,22 @@ object TlsUtils {
 
   }
   case class Certs(caCert: File, serviceCerts: Map[String, ServiceCert])
+
   def withCerts(names: String*)(f: Certs => Unit): Unit = {
+    withCerts(false, names)(f: Certs => Unit)
+  }
+
+  def withCerts(addDnsAltNames: Boolean, names: Seq[String])(f: Certs => Unit): Unit = {
     // First, we create a CA and get a cert/key for linker
     val tmpdir = new File("mktemp -d -t linkerd-tls.XXXXXX".!!.stripLineEnd)
     try {
       val configFile = mkCaDirs(tmpdir)
+      if (addDnsAltNames) {
+        // When we add DNS SAN (Subject Alternative Name) names, it overrides CN (Common Name).
+        // So, tests which modify CN to simulate failure scenarios will start failing if this is added for all.
+        // Reference: https://stackoverflow.com/questions/5935369/ssl-how-do-common-names-cn-and-subject-alternative-names-san-work-together
```

Comment:
If you make `true` the default for `addDnsAltNames` then you can just remove this comment. Instead, add a comment at each `withCerts(false, ...)` call site explaining that the test is explicitly testing Subject CN behavior and not SAN behavior.